src/engraving/infrastructure/eid.cpp

```diff
@@ -19,55 +19,151 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
+#include <string>
+#include <random>
+
+#include <logger.h>
+
 #include "eid.h"
 
 using namespace mu::engraving;
 
-EID::EID(ElementType type, uint32_t id)
-    : m_type(type), m_id(id)
+static constexpr char SEPARATOR = '_';
+
+static std::string int64ToBase64Str(uint64_t n)
 {
-}
+    static constexpr char CHARS[65] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
```

Comment:
Why is that array bigger then the initializer?, it seems `CHARS[64]`, the last element, doesn't get initialized

Reply:
Because C-style strings have the null-termination character

Reply:
Outch... of course. Thanks for getting me straighht ;-)

Reply:
That's true, though as I understand it's not strictly necessary to include the length. This should work too:
```C++
static constexpr char CHARS[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
```